core/src/main/java/com/amplifyframework/core/model/query/predicate/EqualQueryOperator.java

```diff
@@ -40,6 +40,16 @@ public Object value() {
         return value;
     }
 
+    /**
+     * Returns true if the the provided field value equals
+     * the value associated with this operator.
+     * @param field the field value to operate on
+     * @return evaluated result of the operator
+     */
+    public boolean evaluate(Object field) {
+        return field.equals(value);
```

Comment:
what's up with the nullness here? If `value` is non-`null`, then it maybe better to reverse this check and write it as `value.equals(field)`. That way, if `field` is null, you'll get a returned value of `false`, not a `NullPointerException`.

Reply:
Oops. Good catch